biodatasets/tmvar_v3/tmvar_v3.py

```diff
@@ -0,0 +1,306 @@
+                    continue
+        return base_dict
+
+    def pubtator_to_source(self, filepath):
+        """
+        Converts pubtator to source schema
+        """
+        with open(filepath, "r", encoding="utf8") as fstream:
+            for doc in pubtator.iterparse(fstream):
+                document = {}
+                document["pmid"] = doc.pmid
+                title = doc.title
+                abstract = doc.abstract
+                document["passages"] = [
+                    {"type": "title", "text": [title], "offsets": [[0, len(title)]]},
+                    {
+                        "type": "abstract",
+                        "text": [abstract],
+                        "offsets": [[len(title) + 1, len(title) + len(abstract) + 1]],
+                    },
+                ]
+                document["entities"] = [
+                    {
+                        "offsets": [[mention.start, mention.end]],
+                        "text": [mention.text],
+                        "semantic_type_id": [mention.type],
+                        "normalized": self.get_normalizations(mention.id, mention.type),
+                    }
+                    for mention in doc.annotations
+                ]
+                yield document
+
+    def pubtator_to_bigbio_kb(self, filepath):
+        """
+        Converts pubtator to bigbio_kb schema
+        """
+        with open(filepath, "r", encoding="utf8") as fstream:
+            uid = itertools.count(0)
+            for doc in pubtator.iterparse(fstream):
+                document = {}
+                title = doc.title
+                abstract = doc.abstract
+                document["id"] = next(uid)
+                document["document_id"] = doc.pmid
+                document["passages"] = [
+                    {
+                        "id": next(uid),
+                        "type": "title",
+                        "text": [title],
+                        "offsets": [[0, len(title)]],
+                    },
+                    {
+                        "id": next(uid),
+                        "type": "abstract",
+                        "text": [abstract],
+                        "offsets": [[len(title) + 1, len(title) + len(abstract) + 1]],
+                    },
+                ]
+                document["entities"] = [
+                    {
+                        "id": next(uid),
+                        "offsets": [[mention.start, mention.end]],
+                        "text": [mention.text],
+                        "type": [mention.type],
+                        "normalized": self.get_normalizations(mention.id, mention.type),
+                    }
+                    for mention in doc.annotations
+                ]
+                db_id_mapping = {
+                    "dbSNP": "dbSNP",
+                    "CorrespondingGene": "NCBI Gene",
+                    "tmVar": "dbSNP",
```

Comment:
"tmVar" should not be included in the normalization for the `bigbio_kb`, since it's tool-specific. the `db_id` for variants is "dnSNP"

Reply:
It's never going to be "tmVar". This is just a mapping from the source schema to the db-name. If the source schema has "tmVar", it's going to be "dbSNP"

Reply:
```
normalized_bigbio_kb.extend(
                                [
                                    {"db_name": db_id_mapping[key], "db_id": id}
                                    for id in id_list
                                ]
                            )
```
It gets the corresponding db_name from the dictionary and uses that for the `bigbio_kb` schema

Reply:
Maybe I am misunderstanding something. Let me expand on this and you tell me if I got it wrong.

It seems to me that w/ this implementation for this annotation:

```
22051099	425	430	S276T	ProteinMutation p|SUB|S|276|T;CorrespondingGene:3577;RS#:2234671;VariantGroup:0
```

we can end up with:
```python
[
{"db_name" : "dbSNP", "db_id" : "p|SUB|S|276|T"}, 
{"db_name" : "dbSNP", "db_id" : "2234671"},
{"db_name" : "CorrespondingGene", "db_id" : "3577"},
 ]
```
My point here is that in the `bigbio` schema, in the normalization we would like to have only:
```python
[
{"db_name" : "dbSNP", "db_id" : "2234671"},
{"db_name" : "CorrespondingGene", "db_id" : "3577"},
 ]
```

Reply:
Okay, yeah, that makes sense. I was under the impression that the tmVar ID was a more specialized dbSNP id in this normalization scheme. Since it isn't, I've made the necessary change as requested. For CorrespondingGene, the db_name is NCBI Gene in some datasets I saw, so I've maintained that. If it needs to be changed, please let me know